### DISCUSSION TITLE
Next.js App Router Hydration Issue

### DISCUSSION THREAD

**Discussion by aaronbarnaby** [pmndrs/jotai#2067] [Q&A]
When using the app router it seems the hydration isn't working correctly here is my senario. 

I have a shop category page e.g `/category-1`   (with a file structure of `/{categoryName}`
within the layout.tsx I originally used the provided <Provider> to wrap the content, then within a Client Component I used `useHydrateAtoms`.
When I first load this page everything loads fine and the `useAtom` function correctly returns the data.

However If I navigate to `/category-2` when the page first loads it shows the content from `/category-1` then after a short time it changes to the correct content.

I have tried creating my own context using `createStore` and then passing the store when using `useAtom` but this doesn't seem to fix the issue.

My first feeling was it's because the Provider is in the layout file so I tried moving to the `page.tsx` but this didn't seem to have any difference.

Has anyone else had this issue and found any solution?

**Comment by shulcsm** [comment]
You might want to remount `Proivder` providing your categoryName as key if you want to bust store state.
